H4LCppModule.py

```diff
@@ -294,9 +314,13 @@ def analyze(self, event):
             self.worker.SetFsrPhotons(xf.dROverEt2,xf.eta,xf.phi,xf.pt,xf.relIso03)
         for xj in jets:
             self.worker.SetJets(xj.pt,xj.eta,xj.phi,xj.mass,xj.jetId, xj.btagCSVV2, xj.puId)
+
         for xj in FatJets:
             self.worker.SetFatJets(xj.pt, xj.eta, xj.phi, xj.msoftdrop, xj.jetId, xj.btagDeepB, xj.particleNet_ZvsQCD)
-
+	for xg in genparts:
+            self.worker.SetGenParts(xg.pt)
```

Comment:
Also, delete the above two lines that belong to GEN part. It is already there inside `isMC` condition.